### ISSUE TITLE
Docker pull without ":80" in repository name fails permanently on macOS High Sierra after "docker login"

### ISSUE THREAD

**Issue by riy** [moby/moby#37339]
**TL;DR**

Docker for Mac on High Sierra fails to pull an image with a repository tag that does **NOT** contain a port - after the first time `docker login` is performed. Only a _Preferences > Reset_ returns things to normal.

**Description**

I'm running Docker for Mac. I've tried the following with:
- **Docker for Mac EDGE** Version 18.05.0-ce-mac67 (25042) - 0c7f1aa2fa
- Docker for Mac Version 18.03.1-ce-mac65 (24312)
- Docker for Mac Version 17.12.0-ce-mac55 (23011)
 
**Steps to reproduce the issue:**

First of all make sure the image you're testing with is not locally cached but needs to be pulled. Delete them with `docker system prune` and `docker rmi -f` and whatever else you need to do.

1. Sell your first-born, buy a Mac and install High Sierra.
2. Perform a fresh install of one of the above versions or hit _Reset_ in the _Docker Preferences_.
3. Run `docker pull my-repository.domain/path/imagename:version`.
4. Be happy.
5. Run `docker login`.
6. Run `docker pull my-repository.domain/path/imagename`.
7. This will fail with `Error response from daemon: unauthorized: access to the requested resource is not authorized`.
8. Run `docker logout` and pull again. Will still fail.
9. Open the Docker for Mac Preferences and hit _Reset_.
10. Run `docker pull`. This will work, but only until you run `docker login`again, from then on it's step 7.
11. Try all of the above with port `:80` in the repository URL. This will always work fine, regardless if you're logged in or not.

**In a nutshell:**
The `docker pull` command will always work fine if I add `:80` to the URL of the port. But if I remove it (expecting `:80` to be used as a default, which, yes I know, is not how `docker pull` internally works) it will work as long as I don't run `docker login`. Once I do run `docker login` I have to _Reset_ my Docker for Mac installation before I can pull again without specifying the port.

Using Wireshark I can see that it's attempting to connect to port `:443` - and fails at that point.

**Additional information you deem important (e.g. issue happens only occasionally):**

I was able to reproduce this on two different Macs running High Sierra. The problem doesn't occur on El Capitan.

**Output of `docker version`:**

```
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:13:02 2018
 OS/Arch:      darwin/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:22:38 2018
  OS/Arch:      linux/amd64
  Experimental: true
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 18.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.87-linuxkit-aufs
Operating System: Docker for Mac
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.952GiB
Name: linuxkit-025000000001
ID: 67QP:JQDZ:6OSE:Z4UN:D32L:6NTD:RQP4:44E7:MMMV:M46E:ICBF:KAHK
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 21
 Goroutines: 39
 System Time: 2018-06-25T08:47:21.259027135Z
 EventsListeners: 2
HTTP Proxy: docker.for.mac.http.internal:3128
HTTPS Proxy: docker.for.mac.http.internal:3129
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Our Nexus-based Docker repository expects authentication at port `:18449` for pushes, but lets anyone `pull` via ports `:80` and `:8080`. It's also port `:18449` that we run `docker login` against.
 
--

I believe this to be a bug. Any idea what's happening here?

**Comment by riy** [comment]
Closing this issue as I believe this to be a Docker for Mac issue-only. I reposted it in [the appropriate repository](https://github.com/docker/for-mac/issues/3031).
